### PR TITLE
Update macOS DMG background image

### LINKED ISSUE #450: Replace Audacity branding in `mac` directory.
### Discussed in https://github.com/tenacityteam/tenacity/discussions/441

<div type='discussions-op-text'>

<sup>Originally posted by **kirkmarkarian** August  8, 2021</sup>
Just that - I've still got the Audacity headphones for a launch icon.</div>

### PR DESCRIPTION
**Pull request by nbsp**
Signed-off-by: Sol Fisher Romanoff <sol@solfisher.com>

Resolves #450  

Image

This needs some more testing on macOS. 
For some reason, even though I didn't touch the dimensions, it creates a scrollbar to deal with overflowing.

<details>
<summary>Checklist</summary>

- [x] I have signed off my commits using `-s` or `Signed-off-by`\* (See: [Contributing § DCO](https://github.com/tenacityteam/tenacity/blob/master/CONTRIBUTING.md#developer-certificate-of-origin))
- [x] I made sure the code compiles on my machine
- [x] I made sure there are no unnecessary changes in the code\*
- [x] I made sure the title of the PR reflects the core meaning of the issue you are solving\*
- [x] I made sure the commit message(s) contain a description and answer the question "Why do those changes fix that particular issue?" or "Why are those changes really necessary as improvements?"\*

\* indicates required

</details>

### CODE REVIEW

**Comment by n0toose** [comment]
We probably shouldn't put Key over there.

**Comment by nbsp** [comment]
> We probably shouldn't put Key over there.

Blame @Semisol

**Comment by fugidev** [comment]
> For some reason, even though I didn't touch the dimensions, it creates a scrollbar to deal with overflowing.

The same problem actually exists in Audacity's DMG as well. You could make the header a bit more narrow or adjust the window size [here](https://github.com/tenacityteam/tenacity/blob/5e8e48b7b49e3d58c55785e03fff23485c3a0c9f/mac/scripts/build_dist.sh#L246).

Also, we should include a "Retina" (i.e. 144 dpi or @2x) version of the background, it looks blurry on my screen. I read this can be done with a multi-resolution tiff file. 

Regarding key: I don't think it should be in the installer, it might cause confusion. It could be an easter egg in the about window for example, but not here IMO.

**Comment by nbsp** [comment]
> > For some reason, even though I didn't touch the dimensions, it creates a scrollbar to deal with overflowing.
> 
> The same problem actually exists in Audacity's DMG as well. You could make the header a bit more narrow or adjust the window size [here](https://github.com/tenacityteam/tenacity/blob/5e8e48b7b49e3d58c55785e03fff23485c3a0c9f/mac/scripts/build_dist.sh#L246).

Do you know at which resolution it disappears? I don't have a Mac to test.
 
> Also, we should include a "Retina" (i.e. 144 dpi or @2x) version of the background, it looks blurry on my screen. I read this can be done with a multi-resolution tiff file.

I can also just create one high-res file.

> Regarding key: I don't think it should be in the installer, it might cause confusion. It could be an easter egg in the about window for example, but not here IMO.

That's reasonable.

**Comment by Semisol** [comment]
> Blame @Semisol

ouch 😧

**Comment by fitojb** [comment]
I like that it drops unlocalizable text from the image asset. Nice :+1: :

**Comment by nbsp** [comment]
Recreated the image in 2x resolution, please test it on macOS to see if it freaks out

**Comment by nbsp** [comment]
wait, no
it didn't update one sec

**Comment by nbsp** [comment]
Okay, it does break.

Is there a way to make it scale to fit?

**Comment by fugidev** [comment]
Maybe it is required to include the 1x after all, but I'm just guessing.

**Comment by nbsp** [comment]
@FugiMuffi: Can you check if this works on Retina screens?

**Comment by nbsp** [comment]
This is tested and working on my macOS Catalina VM. The overflow scrollbar has been removed.

**Comment by nbsp** [comment]
[This](https://github.com/tenacityteam/tenacity/pull/537/checks?check_run_id=3444312892#step:19:256) doesn't look like the usual race condition error. Dunno why it happens.

image

**Comment by fugidev** [comment]
You are only copying over the 1x version. Also the window size is still the same for me as before, which I don't understand since you did change that.

I combined your two background images into a tiff file using `tiffutil -cathidpicheck background.png background@2x.png -out background.tiff` ([source](https://stackoverflow.com/questions/11199926/create-dmg-with-retina-background-support)), which scales properly in a sample dmg I made using DMG Canvas.

**Comment by nbsp** [comment]
How easy is it to edit the TIFF?

**Comment by nbsp** [comment]
> Also the window size is still the same for me as before, which I don't understand since you did change that.

Does it still have the scrollbar? The reason it overflowed wasn't because of the background, the icons needed a bit more space on the bottom. 8px seemed to be enough.

**Comment by fugidev** [comment]
> How easy is it to edit the TIFF?

It can be imported into gimp as two layers, but editing is not very nice. The tiff could be created on build time instead, that would be more convenient IMO. 

> > Also the window size is still the same for me as before, which I don't understand since you did change that.
> 
> Does it still have the scrollbar? The reason it overflowed wasn't because of the background, the icons needed a bit more space on the bottom. 8px seemed to be enough.

You are right, placing the icons too low still causes overflow, even if the image fits the window.

**Comment by nbsp** [comment]
> It can be imported into gimp as two layers, but editing is not very nice. The tiff could be created on build time instead, that would be more convenient IMO.

I'm a bit out of my depth. @tenacityteam/build-system-gurus, can you help? The branch is open.

**Comment by nbsp** [comment]
Actually, I don't think it matters that much. The files aren't going to be edited individually -- the large file will be downscaled and merged. imo it should be fine to use the TIFF.

File: cmake-proxies/cmake-modules/Package.cmake [hunk 1]
```diff
@@ -41,7 +41,7 @@ if(CMAKE_SYSTEM_NAME MATCHES "Linux")
 elseif( CMAKE_SYSTEM_NAME STREQUAL "Darwin" )
    set( CPACK_GENERATOR DragNDrop )
 
-   set( CPACK_DMG_BACKGROUND_IMAGE "${CMAKE_SOURCE_DIR}/mac/Resources/Audacity-DMG-background.png")
+   set( CPACK_DMG_BACKGROUND_IMAGE "${CMAKE_SOURCE_DIR}/mac/Resources/Tenacity-DMG-background.png")
```
Comment (fugidev) [inline cmake-proxies/cmake-modules/Package.cmake]: you didn't change the file extension here

Comment (nbsp) [reply cmake-proxies/cmake-modules/Package.cmake]: wat, I grepped this

**Comment by nbsp** [comment]
For some reason the DMG generated by the CI still has the scrollbar?

**Comment by fugidev** [comment]
Could it be because of the cache? Maybe someone who knows how that works could take a look

File: scripts/build/macOS/DMGSetup.scpt [hunk 2]
```diff
@@ -7,11 +7,11 @@ on run argv
               set current view of container window to icon view
               set toolbar visible of container window to false
               set statusbar visible of container window to false
-              set the bounds of container window to {400, 100, 1000, 550}
+              set the bounds of container window to {400, 100, 1000, 558}
```
Comment (nbsp) [inline scripts/build/macOS/DMGSetup.scpt]: *AAAAAAAAAAAAAAAAA*

**Comment by nbsp** [comment]
It works! This is ready for review.

**Review by emabrey — APPROVED** [review]

**Pull request closed without merging**